pkg/controllers/managementuser/rbac/namespace_handler.go

```diff
@@ -567,3 +572,26 @@ func updateStatusAnnotation(hasPRTBs bool, namespace *v1.Namespace, mgr *manager
 	}
 
 }
+
+// asyncCleanupRBAC will wait for a Terminating namespace to be fully deleted before removing the associated RBAC.
+func (n *nsLifecycle) asyncCleanupRBAC(namespaceName string) {
+	go func() {
+		for {
+			time.Sleep(cleanupRBACDelaySeconds * time.Second)
```

Comment:
Should we consider using `wait.ExponentialBackoff()` instead?

Reply:
Yeah, I don't mind doing that.  Was mostly tossing this up there to see what folks think of the general approach.  
Should I also add a max for retries as well?

I may also seek some guidance on how to unit test this.

Reply:
looks to be resolved.